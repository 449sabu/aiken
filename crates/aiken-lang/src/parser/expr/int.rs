use chumsky::prelude::*;

use crate::{
    expr::UntypedExpr,
    parser::{error::ParseError, token::Token},
};

pub fn parser() -> impl Parser<Token, UntypedExpr, Error = ParseError> {
    select! { Token::Int {value, base} => (value, base)}.map_with_span(|(value, base), span| {
        UntypedExpr::Int {
            location: span,
            value,
            base,
        }
    })
}
