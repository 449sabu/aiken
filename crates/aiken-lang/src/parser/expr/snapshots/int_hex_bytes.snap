---
source: crates/aiken-lang/src/parser/expr/int.rs
description: "Code:\n\n0x01"
---
Int {
    location: 0..4,
    value: "1",
    base: Hexadecimal,
}
